### PR TITLE
Feature/player movement upgrade

### PR DESCRIPTION
**Pull request by Jokubas126**
* Fire extinguisher

### CODE REVIEW

File: Assets/Scripts/FireEstinguisher.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class FireEstinguisher : MonoBehaviour
```
Comment (latedude2) [inline Assets/Scripts/FireEstinguisher.cs]: missspelled extinguisher :D

File: Assets/Scripts/FireEstinguisher.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    private void Update()
+    {
+        if (Input.GetKeyDown("e") && chargesLeft > 0 && !isShooting && !playerHoldObject.IsHoldingObject)
+        {
+            StartCoroutine(Estinguish());
+        }
+    }
+
+    private IEnumerator Estinguish()
```
Comment (latedude2) [inline Assets/Scripts/FireEstinguisher.cs]: oh no no not again extinguish

File: Assets/Scripts/FireEstinguisher.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    {
+        isShooting = true;
+
+        for(int i=0; i<singleChargeSize; i++)
+        {
+            yield return new WaitForSeconds(0.05f);
+            GameObject waterDrip = Instantiate(waterPrefab, transform.position, Quaternion.identity);
+            waterDrip.GetComponent<Rigidbody>().velocity = transform.TransformDirection(new Vector3(Random.Range(0f, 0.7f), 1, Random.Range(0f, 0.7f)) * chargeVelocity);
```
Comment (latedude2) [inline Assets/Scripts/FireEstinguisher.cs]: variable for spread? 0.7f magic number

File: Assets/Scripts/TileFire.cs [hunk 4]
```diff
@@ -33,6 +33,10 @@ void Update()
             }
             else
             {
+                if (fireParticles.isPlaying && fireResistanceCurrent > 0)
+                {
+                    fireParticles.Stop();
```
Comment (latedude2) [inline Assets/Scripts/TileFire.cs:38]: you might want to set this to pause

**Review by latedude2 — CHANGES_REQUESTED** [review]

File: Assets/Scripts/FireEstinguisher.cs (hunk 1 above)
Comment (Jokubas126) [reply Assets/Scripts/FireEstinguisher.cs]: yeah I forgot that :DD

File: Assets/Scripts/FireEstinguisher.cs (hunk 2 above)
Comment (Jokubas126) [reply Assets/Scripts/FireEstinguisher.cs]: :DDD

File: Assets/Scripts/TileFire.cs (hunk 4 above)
Comment (Jokubas126) [reply Assets/Scripts/TileFire.cs:38]: Mikkel will look this up

**Review by latedude2 — APPROVED** [review]

**Pull request merged**